test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -434,23 +438,70 @@ class test_ucp_sockaddr : public ucp_test {
                                    &recv_length, &params);
     }
 
+    struct rx_am_msg_arg_t {
+        bool received;
+        void *hdr;
+        void *buf;
+
+        rx_am_msg_arg_t(void *_hdr, void *_buf) :
+                received(false), hdr(_hdr), buf(_buf) { }
+    };
+
+    static ucs_status_t rx_am_msg_cb(void *arg, const void *header,
+                                     size_t header_length, void *data,
+                                     size_t length,
+                                     const ucp_am_recv_param_t *param)
+    {
+        volatile rx_am_msg_arg_t *rx_arg =
+                reinterpret_cast<volatile rx_am_msg_arg_t*>(arg);
+        EXPECT_FALSE(rx_arg->received);
+
+        memcpy(rx_arg->hdr, header, header_length);
+        memcpy(rx_arg->buf, data, length);
+
+        rx_arg->received = true;
+        return UCS_OK;
+    }
+
+    void set_am_data_handler(entity &e, uint16_t am_id,
+                             ucp_am_recv_callback_t cb, void *arg)
+    {
+        ucp_am_handler_param_t param;
+
+        /* Initialize Active Message data handler */
+        param.field_mask = UCP_AM_HANDLER_PARAM_FIELD_ID |
+                           UCP_AM_HANDLER_PARAM_FIELD_CB |
+                           UCP_AM_HANDLER_PARAM_FIELD_ARG;
+        param.id         = am_id;
+        param.cb         = cb;
+        param.arg        = arg;
+        ASSERT_UCS_OK(ucp_worker_set_am_recv_handler(e.worker(), &param));
+    }
+
     void send_recv(entity& from, entity& to, send_recv_type_t send_recv_type,
                    bool wakeup, ucp_test_base::entity::listen_cb_type_t cb_type)
     {
         const uint64_t send_data = ucs_generate_uuid(0);
+        uint64_t recv_data       = 0;
+        rx_am_msg_arg_t am_rx_arg(NULL, &recv_data);
         ucs_status_t send_status;
 
+        if (send_recv_type == SEND_RECV_AM) {
+            set_am_data_handler(to, 0, rx_am_msg_cb, &am_rx_arg);
+        }
+
         void *send_req = send(from, &send_data, sizeof(send_data),
                               send_recv_type, scomplete_cbx, NULL);
 
-        uint64_t recv_data = 0;
-        void *recv_req;
+        void *recv_req = NULL;
```

Comment:
@dmitrygx can you pls submit this change, possibly along with adding active messages support, as a separate PR to fix int4 build failures https://dev.azure.com/ucfconsort/ucx_int4/_build/results?buildId=31767&view=logs&j=8cc70074-80e3-506d-e80d-a542b02430e7&t=da02301d-00be-54d1-aa97-9bf0b310a3f6?

Reply:
done #7693